packages/signature-controller/src/SignatureController.test.ts

```diff
@@ -684,6 +691,102 @@ describe('SignatureController', () => {
     });
   });
 
+  describe('trySetDeferredSignSuccess', () => {
+    it('sets a message status as signed in a message manager', () => {
+      signatureController.setDeferredSignSuccess(
+        messageParamsMock.metamaskId,
+        messageParamsMock.data,
+      );
+
+      expect(messageManagerMock.setMessageStatusSigned).toHaveBeenCalledTimes(
+        1,
+      );
+      expect(messageManagerMock.setMessageStatusSigned).toHaveBeenCalledWith(
+        messageIdMock,
+        messageParamsWithoutIdMock.data,
+      );
+
+      expect(
+        personalMessageManagerMock.setMessageStatusSigned,
+      ).not.toHaveBeenCalled();
+      expect(
+        typedMessageManagerMock.setMessageStatusSigned,
+      ).not.toHaveBeenCalled();
+    });
+
+    it('should return false when an error occurs', () => {
+      jest
+        .spyOn(messageManagerMock, 'setMessageStatusSigned')
+        .mockImplementation(() => {
+          throw new Error('mocked error');
+        });
+
+      const result = signatureController.setDeferredSignSuccess(
+        messageParamsMock.metamaskId,
+        messageParamsMock.data,
+      );
+
+      expect(result).toBeUndefined();
+      expect(messageManagerMock.setMessageStatusSigned).toHaveBeenCalledTimes(
+        1,
+      );
+      expect(messageManagerMock.setMessageStatusSigned).toHaveBeenCalledWith(
+        messageIdMock,
+        messageParamsWithoutIdMock.data,
+      );
+    });
+  });
+
+  describe('trySetDeferredSignError', () => {
+    it('rejects a message by calling rejectMessage', () => {
+      signatureController.setDeferredSignError(messageParamsMock.metamaskId);
+
+      expect(messageManagerMock.rejectMessage).toHaveBeenCalledTimes(1);
+      expect(messageManagerMock.rejectMessage).toHaveBeenCalledWith(
+        messageIdMock,
+      );
+
+      expect(personalMessageManagerMock.rejectMessage).not.toHaveBeenCalled();
+      expect(typedMessageManagerMock.rejectMessage).not.toHaveBeenCalled();
+    });
+
+    it('should return false when an error occurs', () => {
```

Comment:
Should this test just verify that it calls `rejectMessage` on the next message manager if the first throws an error?

Perhaps `rejects message on next message manager if first throws`?